src/main/java/io/fabric8/maven/docker/access/hc/DockerAccessWithHcClient.java

```diff
@@ -376,6 +371,26 @@ private HttpBodyAndStatus inspectImage(String name) throws DockerAccessException
         }
     }
 
+    private static Map<ApacheHttpClientDelegate, UrlBuilder> createHttpClientAndUrlBuilder(String baseUrl, String certPath, Integer maxConnections, String apiVersion, Logger log) throws IOException {
```

Comment:
nice trick with the return value, but as we are using commons-lang anyway, you could also use a `Pair<AppacheHttpClientDelegate,UrlBuilder>` and return an `ImmutablePair` here. I think this is more direct (as a map return value does not really make sense here)